src/pages/talks/create/create-talk-form.jsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import axios from 'axios';
+import styled from 'styled-components';
+import { useHistory } from 'react-router-dom';
+import { useFormik } from 'formik';
+
+import { CONFIG } from '../../../constants/config';
+import { Store } from '../../../store/store-provider';
+import { InputField } from '../../../components/input-field';
+import { TextAreaField } from '../../../components/textarea-field';
+import { Button } from '../../../components/button';
+import IcBtnCancel from '../../../assets/ic-btn-cancel.svg';
+import IcBtnAdd from '../../../assets/ic-btn-add.svg';
+const {
+  CreateTalkSchema
+} = require('../../../../server/src/schemas/create-talk-schema');
+
+const FormStyled = styled.form`
+  padding: ${props => props.theme.gridSize}px
+    ${props => props.theme.gridSize * 2}px;
+`;
+
+const ButtonStyled = styled(Button)`
+  margin-left: ${props => props.theme.gridSize * 2}px;
+  padding: 0 ${props => props.theme.gridSize * 2}px 0
+    ${props => props.theme.gridSize}px;
+  height: ${props => props.theme.gridSize * 6}px;
+  border-radius: 2px;
+`;
+
+const ButtonsGroup = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  margin-top: ${props => props.theme.gridSize * 6}px;
+`;
+
+export const CreateTalkForm = () => {
+  const history = useHistory();
+  const { state, dispatch } = React.useContext(Store);
+  const cancel = () => history.push('/');
```

Comment:
Yeah, you’re right. Thanks :)

Reply:
Hey Haishan, This is actually a very interesting topic.

Since I was wrapping functions that I think are "heavy to recreate / heavy to optimize by browsers", but as in JavaScript the creation of a function is pretty lightweight, and this component is not inside of a loop / map, so actually the performance cost is not that big 😁. On the other side, the team might define a code style to have all of us putting all functions inside a useCallback() though.

My latest thoughts are:

1. Performance wise, useCallback() itself does more than just creating a function, so I'd not use it too often if not needed.
2. If it's used inside a loop, either by passing in a function handler or utilize useCallback() to keep it efficient.
3. If dependencies of a useCallback() function changes frequently but it doesn't really matter with the function body, I'd reduce the dependencies or just use barebone functions.

Reply:
Resolved: fixed

Reply:
👍 

Agree that wrap or not wrap is not a big deal in this case.

My logic is simple, I think we should wrap a function with `useCallback` if it will be used as a prop of a child, unless that child is React.memo'ed